### PR TITLE
[PVR] Recordings window: fix crash when selecting '..' (fixes trac#16866)

### PR DESCRIPTION
**Pull request by ksooo**
http://trac.kodi.tv/ticket/16866

Before:
screenshot006

After:
screenshot005

@Jalle19 any objections?
@BigNoid for the (small) Estuary change?
@axbmcuser fyi

### CODE REVIEW

**Comment by Jalle19** [comment]
Wouldn't it be better to just not show the go to parent item if you're already in the root directory?

File: addons/skin.estuary/1080i/MyPVRRecordings.xml [hunk 1]
```diff
@@ -243,12 +243,12 @@
 						<height>262</height>
 						<label>$LOCALIZE[19076] ($INFO[Container(5000).NumItems,[B],[/B] $LOCALIZE[31036]])</label>
 						<font>font16</font>
-						<visible>ListItem.IsFolder</visible>
+						<visible>ListItem.IsFolder + !String.IsEmpty(ListItem.Date)</visible>
```
Comment (ronie) [inline addons/skin.estuary/1080i/MyPVRRecordings.xml]: perhaps use this instead?
```ListItem.IsFolder + !ListItem.IsParentFolder```

**Comment by ksooo** [comment]
> Wouldn't it be better to just not show the go to parent item if you're already in the root directory?

 yes, this would be the best solution, imo. Unfortunately GUIMediaWindow cannot handle empty lists (i was told). There is even really ugly code that adds some ".." (for pvr not really working => root fs listing!) item to prevent problems in case of empty list. => https://github.com/xbmc/xbmc/blob/master/xbmc/windows/GUIMediaWindow.cpp#L1816 ff

File: addons/skin.estuary/1080i/MyPVRRecordings.xml (hunk 1 above)
Comment (ksooo) [inline addons/skin.estuary/1080i/MyPVRRecordings.xml]: Yeah. Much better. Will change.

**Comment by Jalle19** [comment]
Okay, in that case I suppose it's good. Would be nice to be able to fix the root cause somehow though, this path hackery is a bit hard to follow.

**Comment by ksooo** [comment]
> this path hackery is a bit hard to follow.

@Jalle19 reworked. Looks much cleaner to me now. Agreed?

**Comment by ksooo** [comment]
jenkins build this please

**Comment by metaron-uk** [comment]
I suspect this may fix the problem  I was  having
https://github.com/xbmc/xbmc/pull/9336
